commons-core/src/main/scala/com/avsystem/commons/serialization/cbor/CborOutput.scala

```diff
@@ -242,10 +236,13 @@ class CborListOutput(
 
 class CborObjectOutput(
   out: DataOutput,
-  fieldLabels: FieldLabels,
+  keyCodec: CborKeyCodec,
   sizePolicy: SizePolicy
 ) extends CborSequentialOutput(out, sizePolicy) with ObjectOutput {
 
+  private[this] var forcedKeyCodec: CborKeyCodec = _
+  private[this] def currentKeyCodec = if(forcedKeyCodec != null) forcedKeyCodec else keyCodec
```

Comment:
What are the `forcedKeyCodec`s used for? Custom `@cborKey`s and `@cborDiscriminator`s? How do they differ from the `keyCoded` passed through the constructor?

Reply:
The one passed through the constructor is inherited by nested outputs, the "forced" one is only for this particular `Output` instance.